src/internal.c

```diff
@@ -22743,6 +22765,37 @@ int DecodePrivateKey(WOLFSSL *ssl, word16* length)
             if ((*inOutIdx - begin) + dnSz > size)
                 return BUFFER_ERROR;
 
+        #if defined(OPENSSL_ALL) || defined(WOLFSSL_NGINX) || defined(HAVE_LIGHTY)
+            {
+                /* Use a DecodedCert struct to get access to GetName to
+                 * parse DN name */
+                DecodedCert cert;
+                WOLFSSL_X509_NAME* name;
+
+                InitDecodedCert(&cert, input + *inOutIdx, dnSz, ssl->heap);
+
+                if ((ret = GetName(&cert, SUBJECT, dnSz)) != 0) {
+                    FreeDecodedCert(&cert);
+                    return ret;
+                }
+
+                if ((name = wolfSSL_X509_NAME_new()) == NULL) {
+                    FreeDecodedCert(&cert);
+                    return MEMORY_ERROR;
+                }
+
+                CopyDecodedName(name, &cert, SUBJECT);
+
+                if (wolfSSL_sk_X509_NAME_push(ssl->ca_names, name)
+                        == WOLFSSL_FAILURE) {
+                    FreeDecodedCert(&cert);
```

Comment:
Should `name` also be freed in this error case?

Reply:
Yup.